BreakingBad/CharacterListScreen.swift

```diff
@@ -56,46 +56,56 @@ class CharacterListScreen: UIViewController {
     
      
     private func createCharacterList(from characterList : [CharacterJson]) -> [Character] {
```

Comment:
Here is a list of comments focused on this one method:

There is a method hidden in this method, namely  `func character(from character: CharacterJson) -> Character`

i.e you should move the act of creating a single `Character` from a single `CharacterJson` out into its own method. Doing so would leave your original method looking like this: 

```
private func createCharacterList(from characterList: [CharacterJson]) -> [Character] {
    var tempCharacters: [Character] = []
    for character in characterList {
        let character = character(from: character)
        tempCharacters.append(character)
        fetchImage(for: character)
    }
    return tempCharacters
}
```

2. Line 58: no space before a colon `from characterList: [CharacterJson]`

3. What is the purpose of line 61 `let image = defaultImage!` ? The variable image is only used in one place - in the init of `Character` - so instead of creating a dedicated variable for it you could just pass `defaultImage` into the int.

4. line 62: Same comment as point 3. 

5. line 62: Why specify the type of birthday `String`? What does it add? Swift is able to infer type, and your reader doesn’t need to know its type.

6. line 63: is that variable ideally named? Is it actually the `age` or is it `birthdayAndAge` ?

7. line 63: Why define age as a new variable when you could pass `addAgeToBirthday(from: character.birthday)` into the init of `Character` for the age parameter?

8. Line 66+67: You could shorter this to one line: `getImage(for: character)`.  The method `createCharacterList` doesn’t need the URL you extract. And it doesn’t need to know if that fails. All it needs to know is you want to get the image for the character. So move all the URL logic into the `getImage` method. 

As as aside: the call to get the image should not be in this method at all. But we can look at how to move it out in our meeting tomorrow. 

9. Line 67: whether the image is fetched async or synchronously is irrelevant to the purpose of the method, so is irrelevant to the name of the method. You could shorten the method name from `getImageAsync` to `getImage`